### DISCUSSION TITLE
How does one programmatically close a window?

### DISCUSSION THREAD

**Discussion by ChaseLewis** [rust-windowing/winit#2688] [Q&A] [answered]
Having a few issues with winnit but pretty new to Rust. Major issue I'm having currently is I'm trying to encapsulate the API and one thing I want to be able to do is close the application. If the app is full screen I may want to put a button on screen that closes the whole application. I can't find any method to support such an operation which .... is odd from every other window API i've ever worked with. I found a few titles saying 'we already support it but there is no close method' which ... also seemed weird to me. What is the winit way to close a window?

**Comment by kchibisov** [comment, marked as answer]
>Having a few issues with winnit but pretty new to Rust. Major issue I'm having currently is I'm trying to encapsulate the API and one thing I want to be able to do is close the application. If the app is full screen I may want to put a button on screen that closes the whole application. I can't find any method to support such an operation which .... is odd from every other window API i've ever worked with. I found a few titles saying 'we already support it but there is no close method' which ... also seemed weird to me. What is the winit way to close a window?

You `Drop` it, like you do with most other stuff in `Rust` or languages doing `RAII`.

```rust
{
  let window = Window::new(...);

  // <- The window is dropped here, when going out of scope.
}
```
